doc/user/methods/random_forest.md

```diff
@@ -712,3 +717,41 @@ class CustomCategoricalSplit
    dataset will be used to train each decision tree.
  * If `false` _(default for the `ExtraTrees` [variant](#fully-custom-behavior))_, the full
    dataset will be used to train each decision tree.
+ * This parameter will be removed in mlpack 5.0.0. A value of `false` will then be
+   equivalent to `BootstrapType` `IdentityBootstrap` and a value of `true` will be
+   equivalent to `DefaultBootstrap`.
+
+#### `BootstrapType`
+
+ * Specifies the strategy used for bootstrapping data for each .
+ * Three implementations for `BootstrapType` are available for drop-in usage:
+   - `DefaultBootstrap` *(default)*: bootstrap via random sampling with replacement.
+   - `IdentityBootstrap`: no bootstrapping.  Simply copies the input `dataset`, `labels`, and `weights` for each tree's data.
+   - `SequentialBootstrap`: bootstrapping via random sampling with replacement such that samples with informational overlap 
```

Comment:
Now that I understand `SequentialBootstrap` better, we actually have to make some changes here.  I know that they contradict some things I wrote previously; sorry about that.

For `DefaultBootstrap` and `IdentityBootstrap`, it is not necessary to pass an instantiated object to the `RandomForest` for training; they work statically.  So it suffices for a user to just specify `BootstrapType` as a template parameter.

But for `SequentialBootstrap`, the user may need to pass in an instantiated object that contains the data about how samples overlap.  So, we will need to document the constructor and `Train()` variants that allow specifying the instantiated bootstrap object.

Also, we will need to add documentation for the `SequentialBootstrap` itself.  I'm hoping it can be just a couple simple bullet points describing how the sample overlap information is passed.

For clarity it could also be useful to add a simple example usage of `SequentialBootstrap`, which will demonstrate the constructor with all of the arguments and also how to create a `SequentialBootstrap` object.  Even your simple example you used to explain to me could suffice.

Reply:
I updated the documentation to also include an example how to initialize the SequentialBootstrap and subsequently use it for initializing the random forest.